### ISSUE TITLE
Rule proposal: string character access by index

### ISSUE THREAD

**Issue by EvgenyOrekhov** [sindresorhus/eslint-plugin-unicorn#1703]
### Description

There are 3 ways to access string character by index:

1. `string[index]`
2. `string.charAt(index)`
3. `string.at(index)`

I'm guessing one of those ways must be better than the others.
I propose to discuss and decide which one is better.

Maybe we could then create a new rule, or enhance the [`prefer-at`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-at.md) rule (if we decide that `string.at(index)` is the best way).

### Fail

```js
TBD
```

### Pass

```js
TBD
```

**Comment by EvgenyOrekhov** [comment]
I can see that [`prefer-at`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-at.md) rule has `checkAllIndexAccess` option. I guess it already does what I'm talking about, it enforces `string.at(index)`, right?

**Comment by fisker** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/983#issuecomment-853802731

**Comment by fisker** [comment]
No matter which one we choose, we can't know `string[index]` and `string.at(index)` is calling with a string, requires type information.
